### PR TITLE
Custom labels

### PR DESCRIPTION
**Pull request by glennslaven**

### CODE REVIEW

File: metrics.go [hunk 1]
```diff
@@ -82,13 +85,18 @@ func StartReader(file io.Reader, output io.Writer, errorWriter io.Writer) {
 				panic(errors.Wrapf(writeErr, "Writing to output failed"))
 			}
 
-			var logline logLine
+			logline := map[string]string{}
 			jsonErr := json.Unmarshal(line, &logline)
```
Comment (jstangroome) [inline metrics.go]: Does unmarshal fail if any of the values are not strings (i.e. quote-wrapped)?

File: metrics.go [hunk 2]
```diff
@@ -7,26 +7,29 @@ import (
...
-func (l logLine) getBytes() int {
-	bytes, _ := strconv.Atoi(l.Bytes)
+func sanitizeValue(label string, value string) string {
+	switch label {
+	case "content_type":
+		return strings.TrimSpace(strings.Split(value, ";")[0])
+	default:
+		return value
+	}
+}
```
Comment (jstangroome) [inline metrics.go]: impose a max length on everything regardless of label?

File: metrics.go [hunk 3]
```diff
@@ -7,26 +7,29 @@ import (
...
-}
-
 var (
 	filepath string
 )
 
-func (l logLine) getBytes() int {
-	bytes, _ := strconv.Atoi(l.Bytes)
+func sanitizeValue(label string, value string) string {
+	switch label {
```
Comment (jstangroome) [inline metrics.go]: case for `hostname` to strip any port number suffix which has been seen in the wild.

File: metrics.go (hunk 2 above)
Comment (jstangroome) [reply metrics.go]: maybe even trim all leading and trailing whitespace?

Comment (glennslaven) [reply metrics.go]: Any idea on a sane limit. Prometheus doesn't have one, and seems against adding a hard limit at their end: https://github.com/prometheus/prometheus/issues/1235

Comment (glennslaven) [reply metrics.go]: 256 for want of a better number?

Comment (jstangroome) [reply metrics.go]: 256 is probably enough for anybody. I'd go lower if we have data suggesting we can - its easier to allow longer labels in the future than it is to add further constraints.

**Pull request merged**